exec/java-exec/src/main/java/org/apache/drill/exec/store/AbstractSchema.java

```diff
@@ -161,6 +164,30 @@ public boolean showInInformationSchema() {
     return Collections.emptyList();
   }
 
+  /**
+   * Returns a map of types in this schema by name.
+   *
+   * <p>The implementations of {@link #getTypeNames()}
+   * and {@link #getType(String)} depend on this map.
+   * The default implementation of this method returns the empty map.
+   * Override this method to change their behavior.</p>
+   *
+   * @return Map of types in this schema by name
+   */
+  protected Map<String, RelProtoDataType> getTypeMap() {
+    return ImmutableMap.of();
+  }
+
+  @Override
+  public Set<String> getTypeNames() {
```

Comment:
Are these new functions used in the code?

Reply:
In Drill code, they are not used but in Calcite in some places, but it only affects functionality that is not used by Drill yet (CALCITE-2045). Introducing them is required since these methods are added to the `Schema` interface.